pkg/pillar/types/zedagenttypes.go

```diff
@@ -387,52 +387,65 @@ func BootReasonFromString(str string) BootReason {
 }
 
 // MaintenanceModeReason captures reason for entering into maintenance mode
-type MaintenanceModeReason uint8
+type MaintenanceModeReason info.MaintenanceModeReason
+type MaintenanceModeMultyReason []MaintenanceModeReason
 
-// MaintenanceModeReason codes for storing reason for getting into maintenance mode
+// MaintenanceModeReason codes for storing reason for getting into maintenance mode,
+// this should match the values in api/proto/info/info.proto.MaintenanceModeReason
 const (
-	MaintenanceModeReasonNone MaintenanceModeReason = iota
-	MaintenanceModeReasonUserRequested
-	MaintenanceModeReasonVaultLockedUp
-	MaintenanceModeReasonNoDiskSpace
+	MaintenanceModeReasonNone          = MaintenanceModeReason(info.MaintenanceModeReason_MAINTENANCE_MODE_REASON_NONE)
+	MaintenanceModeReasonUserRequested = MaintenanceModeReason(info.MaintenanceModeReason_MAINTENANCE_MODE_REASON_USER_REQUESTED)
+	MaintenanceModeReasonVaultLockedUp = MaintenanceModeReason(info.MaintenanceModeReason_MAINTENANCE_MODE_REASON_VAULT_LOCKED_UP)
+	MaintenanceModeReasonNoDiskSpace   = MaintenanceModeReason(info.MaintenanceModeReason_MAINTENANCE_MODE_REASON_LOW_DISK_SPACE)
+	MaintenanceModeReasonTpmEncFailure = MaintenanceModeReason(info.MaintenanceModeReason_MAINTENANCE_MODE_REASON_TPM_ENCRYPTION_FAILURE)
 )
 
+// String returns the verbose equivalent of MaintenanceModeMultyReason code
+func (mmmr MaintenanceModeMultyReason) String() string {
+	reason := []string{}
+	for _, mmr := range mmmr {
+		reason = append(reason, mmr.String())
+	}
+	return strings.Join(reason, "|")
+}
+
 // String returns the verbose equivalent of MaintenanceModeReason code
 func (mmr MaintenanceModeReason) String() string {
-	switch mmr {
-	case MaintenanceModeReasonNone:
+	if mmr == MaintenanceModeReasonNone {
 		return "MaintenanceModeReasonNone"
-	case MaintenanceModeReasonUserRequested:
+	} else if mmr == MaintenanceModeReasonUserRequested {
 		return "MaintenanceModeReasonUserRequested"
-	case MaintenanceModeReasonVaultLockedUp:
+	} else if mmr == MaintenanceModeReasonVaultLockedUp {
 		return "MaintenanceModeReasonVaultLockedUp"
-	case MaintenanceModeReasonNoDiskSpace:
+	} else if mmr == MaintenanceModeReasonNoDiskSpace {
 		return "MaintenanceModeReasonNoDiskSpace"
-	default:
-		return fmt.Sprintf("Unknown MaintenanceModeReason %d", mmr)
+	} else if mmr == MaintenanceModeReasonTpmEncFailure {
+		return "MaintenanceModeReasonTpmEncFailure"
+	} else {
+		return "Unknown MaintenanceModeReason"
```

Comment:
Can you revert this back to a switch statement? (it is no longer a bitmap which is why you changed it to if/else) AFAIK